File: Samples/ReadUnknownTagValue/C#/ReadUnknownTagValue.cs
using System;
using System.Diagnostics;
using System.Drawing;
using System.IO;

using BitMiracle.LibTiff.Classic;

namespace BitMiracle.LibTiff.Samples
{
    public static class ReadUnknownTagValue
    {
        public static void Main()
        {
            using (Tiff image = Tiff.Open(@"Sample Data\pc260001.tif", "r"))
            {
                // read auto-registered tag 50341
                FieldValue[] value = image.GetField((TiffTag)50341);
                System.Console.Out.WriteLine("Tag value(s) are as follows:");
                for (int i = 0; i < value.Length; i++)
                    System.Console.Out.WriteLine("{0} : {1}", i, value[i].ToString());
            }
        }
    }
}

File: Samples/ReadUnknownTagValue/VB.NET/ReadUnknownTagValue.vb
Imports System
Imports System.Diagnostics
Imports System.Drawing
Imports System.IO

Imports BitMiracle.LibTiff.Classic

Namespace BitMiracle.LibTiff.Samples
    Public NotInheritable Class ReadUnknownTagValue
        Public Shared Sub Main()
            Using image As Tiff = Tiff.Open("Sample Data\pc260001.tif", "r")
                ' read auto-registered tag 50341
                Dim value As FieldValue() = image.GetField(DirectCast(50341, TiffTag))
                System.Console.Out.WriteLine("Tag value(s) are as follows:")
                For i As Integer = 0 To value.Length - 1
                    System.Console.Out.WriteLine("{0} : {1}", i, value(i).ToString())
                Next
            End Using
        End Sub
    End Class
End Namespace
